src/main/java/com/google/sps/data/DatastoreManager.java

```diff
@@ -198,4 +204,44 @@ public static Entity queryForPostEntityWithId(long postId)
         Entity postEntity = (Entity) datastore.get(postEntityKey); 
         return postEntity; 
     }
+
+    /**
+     * Updates a post entity with a reaction
+     * @param postId ID of the post entity 
+     * @param reaction String of the reaction enum 
+     */ 
+    public static void addReactionToPost(long postId, String reaction) 
+    throws EntityNotFoundException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity postEntity = DatastoreManager.queryForPostEntityWithId(postId); 
+        long reactionCount; 
+        if (postEntity.getProperty(reaction) != null) {
+            reactionCount = (long) postEntity.getProperty(reaction); 
+        }
+        else {
+            reactionCount = 0; 
+        }
+        postEntity.setProperty(reaction, reactionCount + 1); 
+        datastore.put(postEntity);
+    }
+
+    /**
+     * Removes a reaction from a post entity 
+     * @param postId ID of the post entity 
+     * @param reaction String of the reaction enum 
+     */ 
+    public static void removeReactionFromPost(long postId, String reaction) 
+    throws EntityNotFoundException, UnsupportedOperationException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Entity postEntity = DatastoreManager.queryForPostEntityWithId(postId); 
+        if (postEntity.getProperty(reaction) == null || (long) postEntity.getProperty(reaction) == 0) {
+            throw new UnsupportedOperationException("No such reactions in post entity"); 
+        }
+        else {
```

Comment:
since the above line returns early, the rest of the function doesn't need to be in a else block

less indentation is more readable